src/Installer/dnup/DotnetInstall.cs

```diff
@@ -2,46 +2,37 @@
 // The .NET Foundation licenses this file to you under the MIT license.
 
 using System;
+using Microsoft.Deployment.DotNet.Releases;
 
 namespace Microsoft.DotNet.Tools.Bootstrapper;
 
-/// <summary>
-/// Base record for .NET installation information with common properties.
-/// </summary>
-public record DotnetInstallBase(
-    string ResolvedDirectory,
+public record DotnetInstallRoot(
+    string? Path,
     InstallType Type,
-    InstallMode Mode,
     InstallArchitecture Architecture)
 {
-    public Guid Id { get; } = Guid.NewGuid();
-}
-
-public record InstallRequestOptions()
-{
-    // Include things such as the custom feed here.
+    //  Do we need a GUID for the ID here?
```

Comment:
I was imagining the GUID could be used to identify unique installs that point to the same component. E.g. my global.json in repo A installed .net 9.0.201 x64 SDK, but repo B also installed it. They might have a different UpdateChannel. The GUID also makes it easier in logs to see what particular component install failed, in my opinion.